change initialization order of CSI and cameras

bitgamma opened the pull request:
`video_mcux_csi_init`, which setups the CSI pins (i.e: calls pinctrl_apply_state) was called after `mt9m114_init` which tries to do i2c communication with the camera to read the chip id. But since one of the CSI pins is the camera master clock, doing things in this order won't work. This PR inverts the order in which the devices are initialized.

also, somewhat related, correct the PWDN pin for pinmux_csi

loicpoulain commented on boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:
```diff
@@ -24,7 +24,7 @@
 	pinmux_csi: pinmux_csi {
 		group0 {
-			pinmux = <&iomuxc_gpio_ad_b0_04_gpio1_io04>;
+			pinmux = <&iomuxc_gpio_ad_b1_02_gpio1_io18>;
```
should be in a different commit.

bitgamma replied:
done

bitgamma committed:
drivers: video: change initialization order of CSI and cameras

video_mcux_csi_init, which setups the CSI pins (i.e: calls
pinctrl_apply_state) was called after mt9m114_init which tries to do i2c
communication with the camera to read the chip id. But since one of the
CSI pins is the camera master clock, doing things in this order won't
work. This PR inverts the order in which the devices are initialized.

Signed-off-by: Michele Balistreri <michele@bitgamma.com>

bitgamma committed:
boards: arm: mimxrt1064_evk: correct PWDN pin of the CSI interface

according to schematics and NXP's sample code, the correct PIN for the
PWDN pin is iomuxc_gpio_ad_b1_02_gpio1_io18

Signed-off-by: Michele Balistreri <michele@bitgamma.com>

loicpoulain reviewed (commented):
Can you please add a commit message explaining the change here.

bitgamma commented:
> Can you please add a commit message explaining the change here.

done!

bitgamma commented:
any chance to get this merged?

teburd commented:
@loicpoulain please re-review when you get a chance, thanks!

dleach02 commented on drivers/video/video_mcux_csi.c:
```diff
@@ -443,6 +445,6 @@ static int video_mcux_csi_init_0(const struct device *dev)
 		    NULL, &video_mcux_csi_data_0,
 		    &video_mcux_csi_config_0,
-		    POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY,
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
```
I would add a comment here to highlight the dependency. Otherwise, future folks may not catch the dependency.

bitgamma replied:
done

dleach02 commented:
@bitgamma please address the compliance issues with your commit message. Look at the history on these files for guidance.

bitgamma commented:
> @bitgamma please address the compliance issues with your commit message. Look at the history on these files for guidance.

done, thanks for the review

danieldegrasse commented on boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:
```diff
@@ -24,7 +24,7 @@
 	/* conflicts with lpuart3 and flexcan1 */
 	pinmux_csi: pinmux_csi {
 		group0 {
-			pinmux = <&iomuxc_gpio_ad_b0_04_gpio1_io04>;
```
Hi @bitgamma, can you update this commit title? Something like the following:

`boards: arm: mimxrt1064_evk: correct PWDN pin of the CSI interface`

Apologies for not catching this earlier

bitgamma replied:
done

mmahadevan108 commented:
@dleach02 @danieldegrasse, can you take a look
